internal/pkg/describe/env.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package describe
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+type EnvironmentSummary struct {
+	Name         string `json:"name"`
+	AccountID    string `json:"accountID"`
+	Region       string `json:"region"`
+	IsProduction bool   `json:"production"`
+}
+
+type Environment struct {
+	*EnvironmentSummary `json:"environment"`
+	Applications        []*Application    `json:"applications"`
+	Tags                map[string]string `json:"tags,omitempty"`
+}
+
+// EnvDescriber retrieves information about an environment.
+type EnvDescriber struct {
+	env  *archer.Environment
+	apps []*archer.Application
+
+	store          envGetter
+	ecsClient      map[string]ecsService
+	stackDescriber stackDescriber
+	sessProvider   sessionFromRoleProvider
+}
+
+// NewEnvDescriber instantiates an environment describer.
+func NewEnvDescriber(projectName string, envName string) (*EnvDescriber, error) {
+	svc, err := store.New()
+	if err != nil {
+		return nil, fmt.Errorf("connect to store: %w", err)
+	}
+	env, err := svc.GetEnvironment(projectName, envName)
+	if err != nil {
+		return nil, err
+	}
+	apps, err := svc.ListApplications(projectName)
+	if err != nil {
+		return nil, err
+	}
+	sess, err := session.NewProvider().FromRole(env.ManagerRoleARN, env.Region)
+	if err != nil {
+		return nil, fmt.Errorf("assuming role for environment %s: %w", env.ManagerRoleARN, err)
+	}
+	return &EnvDescriber{
+		env:            env,
+		store:          svc,
+		apps:           apps,
+		stackDescriber: cloudformation.New(sess),
+		ecsClient:      make(map[string]ecsService),
+		sessProvider:   session.NewProvider(),
+	}, nil
+}
+
+func (e *EnvDescriber) Describe() (*Environment, error) {
+	var envToExpand *EnvironmentSummary
+	envToExpand = &EnvironmentSummary{
```

Comment:
nit: you can use the short-hand to declare a variable :)
```
envToExpand := &EnvironmentSummary {
...
```
https://tour.golang.org/basics/10